### ISSUE TITLE
bug(linux): "Keyman Configuration not responding" message appears on the screen when installing a keyboard

### ISSUE THREAD

**Issue by bharanidharanj** [keymanapp/keyman#10178]
### Describe the bug

While setting up a keyboard, I observed an error message on the screen stating, "Keyman Configuration not responding".

### Reproduce the bug

1. Install Keyman 17.0.225-alpha build on Ubuntu 23.10 mantic Linux OS. 
2. Type "km-config" on the terminal window. 
3. Click the "Download keyboard" button.
4. Type "Khmer" in the Keyboard search bar. 
5.  Select the "Khmer_Angkor" keyboard. 
6. Click the 'Install Keyboard' button. 

I am getting an error message "Keyman Configuration not responding" on the screen. But, the message disappears automatically.  The keyboard is successfully installed afterward. 

..Keyman error message

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/e7d170b7-ec30-4eaf-af30-231b1f39b309">

### Expected behavior

Keyman Configuration error message should not appear during the installation process. 

### Keyman apps

- [ ] Keyman for Android
- [ ] Keyman for iPhone and iPad
- [X] Keyman for Linux
- [ ] Keyman for macOS
- [ ] Keyman for Windows
- [ ] Keyman Developer
- [ ] KeymanWeb
- [ ] Other - give details at bottom of form

### Keyman version

17.0.225-alpha

### Operating system

Ubuntu 23.10 Mantic OS

**Comment by mcdurdin** [comment]
@ermshiperete please retriage as you see fit.

**Comment by ermshiperete** [comment]
Bumping to Future. This can happen if downloading the keyboard package takes too long. We'll have to use async methods to download the keyboard package. Will be fixed with #10338.

**Comment by MAntis127** [comment]
issue is still active even on Ubuntu 25.04 cannot install or remove keyboards without application freezing and displaying '(Not Responding)' Due to issue installing keyboard I cannot use keyman to type in Shavian.

Screenshot (Kubuntu on a Vritual-box machine)

<img width="776" height="455" alt="Image" src="https://github.com/user-attachments/assets/9e46c541-4dff-4be8-b4a2-c3ea3b676802" />

**Comment by ermshiperete** [comment]
@MAntis127 So you're getting the "Keyman configuration not responding" message? If you click the "Wait" button things should proceed. It's possible that the "not responding" message shows up again and that you have to press "Wait" again, but eventually things should succeed.

**Comment by MAntis127** [comment]
how long is two long for a response from the application? 5+ minutes.

<img width="828" height="535" alt="Image" src="https://github.com/user-attachments/assets/da14cce2-b724-46ba-8280-afdff3866367" /> 

Information on virtual machine:
OS: Kubuntu 25.04 Plucky puffin
memory: 16048MB 
3 processers with PAE/NX Nested VT-x/AMD-V

**Comment by mcdurdin** [comment]
> how long is two long for a response from the application? 5+ minutes.

Yes that's definitely too long. I guess something else is going wrong on your system. @ermshiperete how can we diagnose this?

**Comment by ermshiperete** [comment]
Yes, that's way too long. Can you please start km-config with the following command in a terminal window and try to install the keyboard again. Then send me the resulting file `/tmp/km-config.log`.

```
km-config --veryverbose 2>&1 | tee /tmp/km-config.log
```

**Comment by MAntis127** [comment]
[km-config(1).log](https://github.com/user-attachments/files/22435446/km-config.1.log)
 looks like a issue with Ibus. force closed the application after a while so i hope that wont cause any issues.

**Comment by MAntis127** [comment]
@ermshiperete the file is in the comment above

**Comment by ermshiperete** [comment]
Thanks @MAntis127, I'll take a look. Force-closing the app shouldn't cause problems.

**Comment by ermshiperete** [comment]
@MAntis127 I'm trying to track down what might be going wrong and what might be different on your machine since it works in my testing. Could you please try the following commands and then paste the output here:
in a terminal window run `ps -ef | grep ibus`. Then start `km-config` and install the keyboard. While it's still hanging run `ps -ef | grep ibus` again. Then try and run `ibus start -d` (which is what km-config tries to do) - does that work?

**Comment by MAntis127** [comment]
Information on the machine:
<img width="872" height="471" alt="Image" src="https://github.com/user-attachments/assets/cbef2992-e93d-4ed4-99ab-be53a299b271" />

grep of ibus before running km-config
<img width="1305" height="193" alt="Image" src="https://github.com/user-attachments/assets/de732998-c0dd-485c-bdc0-50cfcbf0b6e3" />

grep of ibus after running km-config
<img width="609" height="61" alt="Image" src="https://github.com/user-attachments/assets/ad1d6754-757d-492f-badf-74d24c84a23a" />

@ermshiperete 

ibus is hanging 
<img width="379" height="38" alt="Image" src="https://github.com/user-attachments/assets/ee3efb56-15d7-45e1-b1b7-efd173dce4e8" />

im installing form the web rather than a kmp file

<img width="1886" height="320" alt="Image" src="https://github.com/user-attachments/assets/a241a259-fcd9-4fd2-8d4c-ac50c1ae6de2" />

**Comment by ermshiperete** [comment]
Thanks! I'll dig a bit deeper to see if I can find anything that might cause this. Will be a few days until I get to it, though.

**Comment by MAntis127** [comment]
> Thanks! I'll dig a bit deeper to see if I can find anything that might cause this. Will be a few days until I get to it, though.

Kk i can see that its a kubuntu specific issue. Can sort of install keyboard by turning ibus on and of. but being able to select a keyboard is inconstant sometimes it show up sometimes is just doesn't. Looks like it seem sot work seamlessy on the x11 desktop but insosatanly if at all on wayland.

**Comment by ermshiperete** [comment]
This seems to be a problem in kubuntu or ibus. I can reproduce with just running `ibus start -d` or `ibus exit` on the command line. This also hangs - but not always. (Kubuntu 25.04 Wayland)

**Comment by ermshiperete** [comment]
It seems when `ibus start -d` hangs, I can cancel it. If I then try with `ibus start -d --type=direct` it works, but pops up an IBus notification: "IBus should be called from the desktop session in Wayland. For KDE, you can launch 'systemsettings5' utility and go to "Input Devices" -> "Virtual Keyboard" section and select "IBus Wayland" icon and click "Apply" button to configure IBus in Wayland. For other desktop sessions, you can copy the 'Exec=' line in org.freedesktop.IBus.Panel.Wayland.Gtk3.desktop file to a configuration file of the session. Please refer each document about the "Wayland input method" configuration. Before you configure the "Wayland input method", you should make sure that QT_IM_MODULE and GTK_IM_MODULE environment variables are unset in the desktop session. "

Afterwards `ibus exit` works. And also `ibus start -d` works again.

**Comment by ermshiperete** [comment]
See https://github.com/ibus/ibus/issues/2816

**Comment by MAntis127** [comment]
Thanks for the update hope this continues to be work on. getting Ibus & key-man to function consistently & properly on Wayland

**Comment by ermshiperete** [comment]
The KDE specific problem (reported in https://github.com/keymanapp/keyman/issues/10178#issuecomment-3304755531) will be fixed with PR #14971.

There's still the issue that we should use async functions to download - this will be addressed in #10338.
